docs/pages/ESSupport.md

```diff
@@ -0,0 +1,831 @@
+
+=== "Supported"
+
+    ```javascript
+    sap.ui.define([
+        "Bar"
+    ], 
+    /**
+     * JSDoc block here
+     */
+    (Bar) => Bar.extends("Foo", {
+
+    }));
+    ```
+
+=== "Not supported"
+
+    ```javascript hl_lines="1 2 3"
+    /**
+     * JSDoc block here
+     */
+    sap.ui.define([
+        "Bar"
+    ], (Bar) => Bar.extends("Foo", {
+
+    }));
+    ```
+
+## Code Analysis
+
+While a `ui5 build` is executed, the UI5 Tooling analyses the code on a few places. The following sections explain the analysing parts more detailed.
+
+### Dependency Analysis
+
+The UI5 Tooling analyzes declared dependencies in following places.
+
+#### JSModule Analyzer
+
+The [JSModule Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/JSModuleAnalyzer.js) visiting the AST of a JavaScript file and uses this information to decide whether a code block is executed conditionally or unconditionally. Besides this information which is inherent to the language, the analyzer uses additional knowledge about special APIS / constructs (e.g. the factory function of an AMD module is known to be executed when the module is executed, an IIFE is known to be executed etc.)
+
+#### Component Analyzer
+
+The [Component Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/ComponentAnalyzer.js)  anaylzes JavaScript files named `Component.js` to collect dependency information by searching for a `manifest.json` in the same folder. If it is found and if it is a valid JSON, an "sap.ui5" section is searched and evaluated in the following way
+
+- any library dependency is added as a dependency to the library.js module of that library. If the library dependency is modelled as 'lazy', the
+ module dependency will be added as 'conditional'
+- any component dependency is added as a dependency to the Component.js module of that component. If the Component dependency is modeled as 'lazy', the module dependency will be added as 'conditional'
+- for each configured UI5 module for which a type is configured, a module dependency to that type is added
+- for each route that contains a view name, a module dependency to that view will be added
+
+#### Fiori Elements Analyzer
+
+The [Fiori Elements Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/FioriElementsAnalyzer.js) analyzes a FioriElements app and its underlying template components to collect dependency information. It searches for a `manifest.json` in the same folder. If it is found and if
+it is a valid JSON, an "sap.fe" section is searched and evaluated in the following way
+
+- for each entity set in the "entitySets" object, each sub-entry is checked for a "default"."template" property
+- when found, that string is interpreted as the short name of a template component in package sap.fe.templates
+- a dependency to that template component is added to the analyzed app
```

Comment:
To the component of the app? Can we make a bridge to the previously discussed "Component dependencies"?